internal/proxy/database_interceptor.go

```diff
@@ -0,0 +1,254 @@
+package proxy
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/milvuspb"
+)
+
+// DatabaseInterceptor fill dbname into request based on kv pair <"dbname": "xx"> in header
+func DatabaseInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		filledCtx, filledReq := fillDatabase(ctx, req)
+		return handler(filledCtx, filledReq)
+	}
+}
+
+func fillDatabase(ctx context.Context, req interface{}) (context.Context, interface{}) {
+	switch r := req.(type) {
+	case *milvuspb.CreateCollectionRequest:
```

Comment:
You can try defining an interface with a `GetDbName() string` method. Protobuf messages containing DbName will all have this method, which makes it more extensible.

Reply:
Seem a little tough to unify using the interface, because it also needs a setDBName() method for each request and the generated Protobuf interface doesn't provide it.